airbyte-metrics/metrics-lib/src/main/java/io/airbyte/metrics/lib/OpenTelemetryMetricClient.java

```diff
@@ -75,9 +80,33 @@ public void initialize(MetricEmittingApp metricEmittingApp, String otelEndpoint)
                 .build())
         .setResource(resource)
         .build();
-    OtlpGrpcMetricExporter metricExporter = OtlpGrpcMetricExporter.builder()
+    metricExporter = OtlpGrpcMetricExporter.builder()
         .setEndpoint(otelEndpoint).build();
-    SdkMeterProvider meterProvider = SdkMeterProvider.builder()
+    initialize(metricEmittingApp, sdkTracerProvider, resource);
+  }
+
+  @VisibleForTesting
+  void initializeWithInMemoryExporter(MetricEmittingApp metricEmittingApp) {
+    Resource resource = Resource.getDefault().toBuilder().put(SERVICE_NAME, metricEmittingApp.getApplicationName()).build();
+    metricExporter = InMemoryMetricExporter.create();
+    SdkTracerProvider sdkTracerProvider = SdkTracerProvider.builder()
+        .setResource(resource)
+        .build();
+    initialize(metricEmittingApp, sdkTracerProvider, resource);
+  }
+
+  @VisibleForTesting
+  InMemoryMetricExporter getInMemoryMetricExporter() {
```

Comment:
is there a reason to explicitly return `InMemoryMetricExporter` instead of the generic `MetricExporter`? 

can we make a generic getter function `getMetricExporter` similar to `getSdkMeterProvider`?

Reply:
Sure - moved the casting in the test class